src/main/java/rssreader/dao/impl/RssDAOImpl.java

```diff
@@ -0,0 +1,67 @@
+package rssreader.dao.impl;
+
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+import rssreader.dao.RssDAO;
+import rssreader.entity.RssEntity;
+import rssreader.util.HibernateUtil;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class RssDAOImpl implements RssDAO{
+
+    @Override
+    public List<RssEntity> getAllNews(long feedId) {
+        List<RssEntity> news;
+        try(Session session = HibernateUtil.getSessionFactory().openSession()){
+            Query query = session.createQuery("FROM RssEntity WHERE feed_id = :feedId")
+                    .setParameter("feedId", feedId);
+            news = query.list();
+        }
+        return news;
+    }
+
+    @Override
+    public List<RssEntity> getNewsPage(long feedId, int start, int end) {
+        List<RssEntity> news;
+        try(Session session = HibernateUtil.getSessionFactory().openSession()){
+            Query query = session.createNativeQuery("SELECT newsId, newsTitle, newsLink, newsDescription, newsGuid, newsPubDate " +
+                    "FROM (SELECT ROWNUM rnum, news.NEWS_ID as newsId, news.NEWS_TITLE as newsTitle, news.NEWS_LINK as newsLink, news.NEWS_DESCRIPTION as newsDescription, news.NEWS_GUID as newsGuid, news.NEWS_PUBDATE as newsPubDate " +
+                    "FROM (SELECT * FROM RSSREADER.NEWS WHERE FEED_ID = " + feedId + ") news " +
+                    "WHERE ROWNUM <= " + end + ") " +
+                    "WHERE rnum >= " + start);
+            news = ConvertObjectList(query.list());
+        }
+        return news;
+    }
+
+    @Override
+    public Optional<RssEntity> getNewsById(long id) {
```

Comment:
The method name is not appropriate: this one return Feed, not News, isn't it? getFeedById()